tests/rptest/tests/audit_log_test.py

```diff
@@ -588,6 +591,15 @@ def get_metrics_from_node_sync(patterns: list[str]):
             "audit_client_buffer_usage_ratio",
         ]
 
+        # wait until `audit_client_buffer_usage_ratio` appears
+        # on every node
+        # i.e. until audit client is fully initialized
+        for node in self.redpanda.nodes:
+            wait_until(
```

Comment:
nit: You can use `wait_until_result` to grab the result instead of having to call `get_metrics_from_node` in another for loop over the same set again. Maybe something like:
```
def my_ex_fn():
    wait_until(has_metric)
    return get_metrics_from_node(node, metrics)
    
for node in self.redpanda.nodes:
    samples = my_ex_fn()
    assert samples
```